tasks/func.py

```diff
@@ -114,6 +114,13 @@ def invoke(ctx, user, func, input_data=None):
     if input_data:
         data["input_data"] = input_data
 
+    if mpi is not None:
+        data["mpi_world_size"] = int(mpi)
+
+    if graph:
+        data["record_exec_graph"] = True
+        data["async"] = True
```

Comment:
Some times (e.g. LAMMPS) we want to always run with the `async` flag on, even if we are not recording the execution graph. Feel free to add another header flag if you think it can be necessary.

Reply:
Yeah interesting, I'm not really sure how/ whether we should have one place for all the function invocation logic, or have it spread around. It feels like the Faasm API is stable enough now that this should really all be contained in a PyPi package for `faasm` so we could import it in here and in the experiments repos